### PR TITLE
cleanup: Refactor RandomDatabaseName() function.

### PR DESCRIPTION
**Pull request by coryan**
I am going to create a new integration test, and want to use this
function in it. Refactor the function to the testing library.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp-spanner/292)

### CODE REVIEW

File: google/cloud/spanner/testing/random_database_name.cc [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  // A database ID must be between 2 and 30 characters, fitting the regular
+  // expression `[a-z][a-z0-9_\-]*[a-z0-9]`
+  int max_size = 30;
+  std::string const prefix = "db-";
+  auto size = static_cast<int>(max_size - 1 - prefix.size());
+  return prefix +
+         google::cloud::internal::Sample(
+             generator, size, "abcdefghijlkmnopqrstuvwxyz012345689_-") +
+         google::cloud::internal::Sample(generator, 1,
+                                         "abcdefghijlkmnopqrstuvwxyz");
```
Comment (devbww) [inline google/cloud/spanner/testing/random_database_name.cc]: Nit: The regex above says we can have digits here.  Should we try to generate such?

File: google/cloud/spanner/testing/random_database_name.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+#include "google/cloud/spanner/version.h"
+#include "google/cloud/internal/random.h"
+
+namespace google {
+namespace cloud {
+namespace spanner_testing {
+inline namespace SPANNER_CLIENT_NS {
+
+/// Create a random database name given a PRNG generator.
+std::string RandomDatabaseName(google::cloud::internal::DefaultPRNG& generator);
```
Comment (devbww) [inline google/cloud/spanner/testing/random_database_name.h]: `#include <string>`

File: google/cloud/spanner/testing/random_database_name.cc (hunk 1 above)
Comment (remyabel) [reply google/cloud/spanner/testing/random_database_name.cc]: The prefix contains `db-` but I don't think the regex has a dash between `[a-z]` and `[a-z0-9_\-]`.

Comment (remyabel) [reply google/cloud/spanner/testing/random_database_name.cc]: Nevermind, disregard my last comment.

Comment (coryan) [reply google/cloud/spanner/testing/random_database_name.cc]: Done.

File: google/cloud/spanner/testing/random_database_name.h (hunk 2 above)
Comment (coryan) [reply google/cloud/spanner/testing/random_database_name.h]: Done.

File: google/cloud/spanner/testing/random_database_name.cc [hunk 3]
```diff
@@ -29,8 +28,8 @@ std::string RandomDatabaseName(
   return prefix +
          google::cloud::internal::Sample(
              generator, size, "abcdefghijlkmnopqrstuvwxyz012345689_-") +
-         google::cloud::internal::Sample(generator, 1,
-                                         "abcdefghijlkmnopqrstuvwxyz");
+         google::cloud::internal::Sample(
+             generator, 1, "abcdefghijlkmnopqrstuvwxyz01234567890");
```
Comment (devbww) [inline google/cloud/spanner/testing/random_database_name.cc]: It looks like there is an extra zero.

Comment (coryan) [reply google/cloud/spanner/testing/random_database_name.cc]: Fixed.

**Pull request merged**